core/templates/pages/skill-editor-page/editor-tab/skill-concept-card-editor/worked-example-editor.component.spec.ts

```diff
@@ -0,0 +1,148 @@
+import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
+import { SkillUpdateService } from 'domain/skill/skill-update.service';
+import { WorkedExampleEditorComponent } from './worked-example-editor.component';
+
+describe('Worked example editor component', () => {
+  let component: WorkedExampleEditorComponent;
+  let fixture: ComponentFixture<WorkedExampleEditorComponent>;
+  let skillUpdateService: SkillUpdateService;
+
+  beforeEach(waitForAsync(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+      declarations: [
+        WorkedExampleEditorComponent
+      ],
+      providers: [
+        ChangeDetectorRef,
+        SkillUpdateService
+      ],
+      schemas: [NO_ERRORS_SCHEMA]
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(WorkedExampleEditorComponent);
+    component = fixture.componentInstance;
+    skillUpdateService = TestBed.inject(SkillUpdateService);
+
+    component.isEditable = true;
+    component.index = 2;
+    component.workedExample = {
+      getQuestion(): object {
+        return {
+          html: 'worked example question 1',
+          content_id: 'worked_example_q_1',
+        };
+      },
+
+      getExplanation(): object {
+        return {
+          html: 'worked example explanation 1',
+          content_id: 'worked_example_e_1',
+        };
+      }
+    };
+    component.ngOnInit();
+  });
+
+  it('should set properties when initialized', () => {
+    expect(component.questionEditorIsOpen).toBe(false);
+    expect(component.explanationEditorIsOpen).toBe(false);
+    expect(component.WORKED_EXAMPLE_FORM_SCHEMA).toEqual({
+      type: 'html',
+      ui_config: {}
+    });
+  });
+
+  it('should open question editor when clicking on edit button', () => {
+    expect(component.questionEditorIsOpen).toBe(false);
+
+    component.openQuestionEditor();
+
+    expect(component.questionEditorIsOpen).toBe(true);
+  });
+
+  it('should open explanation editor wclicking on edit button', () => {
+    expect(component.explanationEditorIsOpen).toBe(false);
+
+    component.openExplanationEditor();
+
+    expect(component.explanationEditorIsOpen).toBe(true);
+  });
+
+  it('should close question editor when clicking on cancel button', () => {
+    component.openQuestionEditor();
+
+    component.cancelEditQuestion();
+
+    expect(component.questionEditorIsOpen).toBe(false);
+  });
+
+  it('should close explanation editor when clicking on cancel button', () => {
+    component.openExplanationEditor();
+
+    component.cancelEditExplanation();
+
+    expect(component.explanationEditorIsOpen).toBe(false);
+  });
+
+  it('should save worked example when clicking on save button', () => {
+    let skillUpdateSpy = spyOn(skillUpdateService, 'updateWorkedExample')
+      .and.returnValue(null);
+
+    component.saveWorkedExample(true);
+
+    expect(skillUpdateSpy).toHaveBeenCalled();
```

Comment:
Can you use `toHaveBeenCalledWith()` instead?

Reply:
Done.